src/components/tl/ucp/tl_ucp_ep.h

```diff
@@ -70,10 +80,10 @@ static inline ucc_status_t ucc_tl_ucp_get_ep(ucc_tl_ucp_team_t *team, ucc_rank_t
             *ep = NULL;
             return status;
         }
-        if (ctx->eps) {
-            ctx->eps[ctx_rank] = *ep;
+        if (team->worker->eps) {
+            team->worker->eps[ctx_rank] = *ep;
         } else {
-            tl_ucp_hash_put(ctx->ep_hash, h->ctx_id, *ep);
+            tl_ucp_hash_put(team->worker->ep_hash, h->ctx_id, *ep);
```

Comment:
@samnordmann plz see linter warnings here: https://github.com/openucx/ucc/actions/runs/3082772787/jobs/4982882880

here linter has a false positive: it assumes "h" can be NULL at line 86. Either use annotation to suppress or try to rewrite the same code a bit (maybe try "if (!h) " instead of "if (team->worker->eps)" at line 83).

Another warning is tl_ucp_context.c line 275: ucc_free(prefix) - double free. This is my fault (the PR we merged into your branch) - should be a simple fix (e.g. set prefix ot NULL after free and check in the end)

Reply:
done. Btw I fixed another issue: `prefix `was freed and then passed as an argument to `ucc_tl_ucp_context_service_init `